_stbt/control_gpl.py

```diff
@@ -1,16 +1,72 @@
+from .logging import debug
+
+
+class HdmiCecError(Exception):
+    pass
+
+
 class HdmiCecControl(object):
+
+    # Map our recommended keynames (from linux input-event-codes.h) to the
+    # equivalent CEC commands.
+    # The mapping between CEC commands and code can be found at http://www.cec-o-matic.com
+    # or in HDMI-CEC specification 1.3a
+    _KEYNAMES = {
+        "KEY_OK": "14:44:00",
+        "KEY_UP": "14:44:01",
+        "KEY_DOWN": "14:44:02",
+        "KEY_LEFT": "14:44:03",
+        "KEY_RIGHT": "14:44:04",
+        "KEY_BACK": "14:44:0D"
+    }
+
     def __init__(self, device):
```

Comment:
To make the source and destinations configurable make this

    def __init__(self, device=None, source=None, destination=None):

and add

    if source is None:
        source = 0
    if destination is None:
        destination = 0xf

here